config/routes.rb

```diff
@@ -15,6 +15,4 @@
   resource :order do
```

Comment:
Why are you ignoring my comments?
Why do you choose to use a single resource for orders? Is your website can have only one order?
Please fix it, rename or describe your choose.

Reply:
Because we don't need to get access to any order but only one current_order. I think we don't need url like this /orders/ID/order_items

Reply:
Please explain me why not and I'll change this. But I can't do something without understanding why. 
Some time ago you send me a link where was explanation about resources and single resource. So why we can't realize there single resource? 
The same question can be asked for resources: users

Reply:
I suppose you have to separate these routes
If your system or resource has more than one resource it should be `resources`
```
  resources :users do
    resources :orders
  end
```
If your system or resource has only one resource it should be `resource`
```
  resources :users do
    resource :billing_address
  end
```
About `/orders/ID/order_items`
In your case, when you use OrdersItemsController#index the routes have to be 
```
  resources :orders do
   resources :order_items
```
You can create another controller for managing your cart (the order without saving).

About orders. 
Can only one `order` be saved in your database? Can your users have more than one `order`?

About users.
Can only one `user` be saved in your database?